File: backend/tests/DigitalFamilyCookbook.Tests.Utilities/Mocks/MockMeat.cs
namespace DigitalFamilyCookbook.Tests.Utilities.Mocks;

public static class MockMeat
{
    public static MeatDto GenerateMeatDto() => new MeatDto
    {
        MeatId = MockDataGenerator.RandomInteger(),
        Id = MockDataGenerator.RandomId(),
        Name = MockDataGenerator.RandomString(10, false),
    };

    public static Meat GenerateMeat() => Meat.FromDto(GenerateMeatDto());

    public static List<Meat> GenerateMeatList(int count = -1)
    {
        var meatCount = count;

        if (meatCount < 0)
        {
            meatCount = MockDataGenerator.RandomInteger(1, 5);
        }

        var meats = new List<Meat>();

        for (var i = 0; i < meatCount; i += 1)
        {
            meats.Add(GenerateMeat());
        }

        return meats;
    }
}

File: frontend/src/test/mocks/MockMeat.ts
import DataGenerator from '@test/DataGenerator';
import { Meat } from '@models/Meat';

export const MockMeat = (id = 0): Meat => ({
    id: DataGenerator.GenerateGuid(),
    meatId: id === 0 ? DataGenerator.GenerateInteger(1, 250) : id,
    name: DataGenerator.GenerateString(),
});

export const MockMeatList = (length = 10): Meat[] => {
    const meats: Meat[] = [];

    for (let i = 0; i < length; i += 1) {
        meats.push(MockMeat(i + 1));
    }

    return meats;
};
